exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -197,32 +197,19 @@ func (m *AzureManagedMachinePool) ValidateUpdate(oldRaw runtime.Object, client c
 		}
 	}
 
-	if old.Spec.EnableUltraSSD != nil {
-		// Prevent EnabledUltraSSD modification if it was already set to some value
-		if m.Spec.EnableUltraSSD == nil {
-			// unsetting the field is not allowed
-			allErrs = append(allErrs,
-				field.Invalid(
-					field.NewPath("Spec", "EnableUltraSSD"),
-					m.Spec.EnableUltraSSD,
-					"field is immutable, unsetting is not allowed"))
-		} else if *m.Spec.EnableUltraSSD != *old.Spec.EnableUltraSSD {
-			// changing the field is not allowed
-			allErrs = append(allErrs,
-				field.Invalid(
-					field.NewPath("Spec", "EnableUltraSSD"),
-					m.Spec.EnableUltraSSD,
-					"field is immutable"))
-		}
-	} else {
-		if m.Spec.EnableUltraSSD != nil {
-			allErrs = append(allErrs,
-				field.Invalid(
-					field.NewPath("Spec", "EnableUltraSSD"),
-					m.Spec.EnableUltraSSD,
-					"field is immutable, unsetting is not allowed"))
-		}
+	if err := validateBoolPtrImmutable(
+		field.NewPath("Spec", "EnableUltraSSD"),
+		old.Spec.EnableUltraSSD,
+		m.Spec.EnableUltraSSD); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if err := validateBoolPtrImmutable(
```

Comment:
I don't actually see specific documentation that says that this node pool configuration is create-time only.

https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools#assign-a-public-ip-per-node-for-your-node-pools

Did we confirm that you definitely can't change this configuration after the node pool is already created?

Reply:
Yeah, when I try to edit that field I see errors in the logs like
```
E0908 20:44:39.439710     103 controller.go:326]  "msg"="Reconciler error" "error"="error creating AzureManagedMachinePool default/pool1: failed to reconcile machine pool pool1: failed to create resource aks-jon/pool1 (service: agentpools): failed to begin operation: Code=\"PropertyChangeNotAllowed\" Message=\"Changing property 'agentPoolProfile.enableNodePublicIP' is not allowed.\" Target=\"agentPoolProfile.enableNodePublicIP\"" "azureManagedMachinePool"={"name":"pool1","namespace":"default"} "controller"="azuremanagedmachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureManagedMachinePool" "name"="pool1" "namespace"="default" "reconcileID"="5c493f37-3585-40b3-ac0a-b93b548ab047" 
```

Reply:
That same error shows up in both cases when it was created as true or false and toggling to the other.

Reply:
Great, that settles it!